utils/predicate/predicate_bytes.go

```diff
@@ -0,0 +1,68 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package predicateutils
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// PredicateEndByte is used as a delimiter for the bytes packed into a precompile predicate.
+// Precompile predicates are encoded in the Access List of transactions in the access tuples
+// which means that its length must be a multiple of 32 (common.HashLength).
+// For messages with a length that does not comply to that, this delimiter is used to
+// append/remove padding.
+var PredicateEndByte = byte(0xff)
+
+// PackPredicate packs [predicate] by delimiting the actual message with [PredicateEndByte]
+// and zero padding to reach a length that is a multiple of 32.
+func PackPredicate(predicate []byte) []byte {
+	predicate = append(predicate, PredicateEndByte)
+	return common.RightPadBytes(predicate, (len(predicate)+31)/32*32)
+}
```

Comment:
Won't the final predicate byte length be  (len(predicate) + some multiple of 32), instead of what we want which is some multiple of 32?